core/src/components/select/select.tsx

```diff
@@ -251,32 +257,48 @@ export class Select implements ComponentInterface {
   }
 
   private createAlertInputs(data: HTMLIonSelectOptionElement[], inputType: 'checkbox' | 'radio', selectValue: any): AlertInput[] {
-    return data.map(o => {
-      const value = getOptionValue(o);
+    const alertInputs = data.map(option => {
+      const value = getOptionValue(option);
+
+      // Remove hydrated before copying over classes
+      option.classList.remove('hydrated');
```

Comment:
Why do we need to remove the hydrated class?

Reply:
We technically don't, but it just seems odd passing it to a non-stencil component. It's only needed for stencil component visibility.